### DISCUSSION TITLE
Tags, Checkboxes, Chips, Radio-buttons, Pills, Selected Buttons, etc

### DISCUSSION THREAD

**Discussion by turolopezsanabria** [SUI-Components/sui-components#2103] [Ideas and new features]
### Status of this discussion

* [x] Ongoing
* [ ] AGREED (We have an agreement)
	* [ ] New Component
	* [ ] New Feature
	* [ ] Other 
* [ ] DONE (Closed, Discarded or Implemented)

# Description

> You can link your code or Figma file here

As started in [this weekly Sync on Twitch](https://www.twitch.tv/videos/1447885711) ...

## Constraints and Blockers 

* a
* b
* …

## Proposals / Options

* Option A
	* Pros:
	* Cons:
* Option B
	* Pros:
	* Cons:
* ...

## Related links 

* Article [What's the difference between pills chips...](https://ux.stackexchange.com/questions/98069/what-is-the-difference-between-pills-chips-and-tags-tokens/141416) 
* 

# Decision / Outcome of this discussion

## Rationale

> i.e. We've decided to continue with option "A" because [postitive arguments] while "B" [negative arguments]

**Comment by andresin87** [comment]
# v1.2 current (nov21)  https://www.w3.org/TR/wai-aria-practices/
# v1.3 draft (mar22) https://w3c.github.io/aria/

**Reply by andresin87** [reply]
# Button 
## aria 1.2 (current)

https://www.w3.org/TR/wai-aria-practices/#button

A [button](https://www.w3.org/TR/wai-aria-1.2/#button) is a widget that enables users to trigger an action or event, such as submitting a form, opening a dialog, canceling an action, or performing a delete operation. A common convention for informing users that a button launches a dialog is to append "…" (ellipsis) to the button label, e.g., "Save as…".

In addition to the ordinary button widget, WAI-ARIA supports 2 other types of buttons:

- Toggle button: A two-state button that can be either off (not pressed) or on (pressed). To tell assistive technologies that a button is a toggle button, specify a value for the attribute [aria-pressed](https://www.w3.org/TR/wai-aria-1.2/#aria-pressed). For example, a button labelled mute in an audio player could indicate that sound is muted by setting the pressed state true. Important: it is critical the label on a toggle does not change when its state changes. In this example, when the pressed state is true, the label remains "Mute" so a screen reader would say something like "Mute toggle button pressed". Alternatively, if the design were to call for the button label to change from "Mute" to "Unmute," the aria-pressed attribute would not be needed.
- Menu button: as described in the [menu button pattern](https://www.w3.org/TR/wai-aria-practices/#menubutton), a button is revealed to assistive technologies as a menu button if it has the property [aria-haspopup](https://www.w3.org/TR/wai-aria-1.2/#aria-haspopup) set to either menu or true.


#### NOTE
The types of actions performed by buttons are distinctly different from the function of a link (see [link pattern](https://www.w3.org/TR/wai-aria-practices/#link)). It is important that both the appearance and role of a widget match the function it provides. Nevertheless, elements occasionally have the visual style of a link but perform the action of a button. In such cases, giving the element role button helps assistive technology users understand the function of the element. However, a better solution is to adjust the visual design so it matches the function and ARIA role.

<img src="https://img.tecnomagazine.net/2017/06/ring-timbre-inteligente.jpg" width="200"/>

#### ButtonGroup
<img src="https://user-images.githubusercontent.com/1674036/162172330-351eb304-d8aa-4d5a-97d7-ee8f12136930.jpg" width="200" /> <img src="https://user-images.githubusercontent.com/1674036/162172868-b962cbe5-1771-4c7f-a309-bf17d7d4e532.jpeg" width="200" /> <img src="https://user-images.githubusercontent.com/1674036/162173031-027040b2-b72c-4a06-b6f5-53eaeb89d9e6.jpg" width="200" />

**Reply by andresin87** [reply]
# Checkbox
## aria 1.2 current
https://www.w3.org/TR/wai-aria-practices/#checkbox

WAI-ARIA supports two types of [checkbox](https://www.w3.org/TR/wai-aria-1.2/#checkbox) widgets:

- Dual-state: The most common type of checkbox, it allows the user to toggle between two choices -- checked and not checked.
- Tri-state: This type of checkbox supports an additional third state known as partially checked.

One common use of a tri-state checkbox can be found in software installers where a single tri-state checkbox is used to represent and control the state of an entire group of install options. And, each option in the group can be individually turned on or off with a dual state checkbox.

- If all options in the group are checked, the overall state is represented by the tri-state checkbox displaying as checked.
- If some of the options in the group are checked, the overall state is represented with the tri-state checkbox displaying as partially checked.
- If none of the options in the group are checked, the overall state of the group is represented with the tri-state checkbox displaying as not checked.
The user can use the tri-state checkbox to change all options in the group with a single action:

- Checking the overall checkbox checks all options in the group.
- Unchecking the overall checkbox will uncheck all options in the group.
- And, In some implementations, the system may remember which options were checked the last time the overall status was partially checked. If this feature is provided, activating the overall checkbox a third time recreates that partially checked state where only some options in the group are checked.


#### ToogleButtonGroup as CheckboxGroup
<img src="https://user-images.githubusercontent.com/1674036/162171838-51c37904-643f-413d-8812-ef2cae9335cd.jpeg" width="200px" />

**Reply by andresin87** [reply]
# RadioGroup
## aria 1.2 current
https://www.w3.org/TR/wai-aria-practices/#radiobutton

A radio group is a set of checkable buttons, known as radio buttons, where no more than one of the buttons can be checked at a time. Some implementations may initialize the set with all buttons in the unchecked state in order to force the user to check one of the buttons before moving past a certain point in the workflow.


#### ToogleButtonGroup as RadioButtonGroup
<img src="https://user-images.githubusercontent.com/1674036/162167982-fd052080-7753-4bb6-80ba-8debd8704e2c.jpeg" width="200" />
